http/query_handler.go

```diff
@@ -179,6 +175,51 @@ func (h *FluxHandler) handleQuery(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// handleFluxError will take a flux.Error and convert it into an influxdb.Error.
+// It will match certain codes to the equivalent in influxdb.
+//
+// If the error is any other type of error, it will return the error untouched.
+//
+// TODO(jsternberg): This likely becomes a public function, but this is just an initial
+// implementation so playing it safe by making it package local for now.
+func handleFluxError(err error) error {
+	ferr, ok := err.(*flux.Error)
+	if !ok {
+		return err
+	}
+
+	code := influxdb.EInternal
+	switch flux.ErrorCode(err) {
+	case codes.NotFound:
+		code = influxdb.ENotFound
+	case codes.Invalid:
+		code = influxdb.EInvalid
+	// These don't really map correctly, but we want
+	// them to show up as 4XX so until influxdb error
+	// codes are updated for more types of failures,
+	// mapping these to invalid.
+	case codes.Canceled,
+		codes.ResourceExhausted,
+		codes.FailedPrecondition,
+		codes.Aborted,
+		codes.OutOfRange,
+		codes.Unimplemented:
+		code = influxdb.EInvalid
+	case codes.PermissionDenied:
+		code = influxdb.EForbidden
+	case codes.Unauthenticated:
+		code = influxdb.EUnauthorized
```

Comment:
Seems weird to match unauthenticated to unauthorized since they are different but easily confused.  I'm not sure what it should map to instead though.

Reply:
It's because these are very closely related to the HTTP status codes where the 401 unauthorized basically means "you have not authenticated" while 403 forbidden is "you authenticated, but you don't have permission".